group/g1.go

```diff
@@ -45,6 +47,26 @@ func (z *G1Element) AddAssign(e Element) Element {
 	return z
 }
 
+func (z *G1Element) Add(a, b Element) Element {
+	aa := a.(*G1Element).inner
+	bb := b.(*G1Element).inner
+	z.inner.Set(&aa)
+	z.inner.AddAssign(&bb)
+	return z
+}
+
+func (z *G1Element) MultiExp(base []Element, scalars []fr.Element) (Element, error) {
+	jacs := make([]bls12381.G1Jac, len(base))
+	for i := 0; i < len(base); i++ {
+		jacs[i] = base[i].(*G1Element).inner
+	}
+	affs := bls12381.BatchJacobianToAffineG1(jacs)
+	if _, err := z.inner.MultiExp(affs, scalars, common.MultiExpConf); err != nil {
+		return nil, fmt.Errorf("g1 multiexp: %s", err)
+	}
+	return z, nil
+}
```

Comment:
For G1 we can leverage gnark-crypto implementation for MSM which should be quite efficient. (i.e: is what we did before when we only supported G1).